modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache

```diff
@@ -175,7 +175,7 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
     {{/hasFormParams}}
     {{#bodyParam}}
             const needsSerialization = (<any>"{{dataType}}" !== "string") || localVarRequestOptions.headers['Content-Type'] === 'application/json';
-            localVarRequestOptions.data =  needsSerialization ? JSON.stringify({{paramName}} || {}) : ({{paramName}} || "");
+            localVarRequestOptions.data =  needsSerialization ? JSON.stringify({{paramName}} !== undefined ? {{paramName}} : {}) : ({{paramName}} || "");
```

Comment:
is it correct to send an empty object when the body is undefined?
what if body is null?

Reply:
`null` is a valid JSON value

For `{}` when it's `undefined`, I have no idea so I kept the current behavior

Reply:
But you're right, an `undefined` body param should either throw an error or give an empty body :thinking:

Reply:
I think it is ok to treat `undefined` similar to `null` and send an empty object.

Reply:
the more I think about it, the more I consider the `const needsSerialization = (<any>"{{dataType}}" !== "string")` part the origin of the problem, since it does not check for any other type like boolean, etc.
so the `undefined` check is a good way to send whatever is passed as body as long as it is defined